src/osd/OSD.cc

```diff
@@ -7986,6 +8005,42 @@ void OSD::trim_maps(epoch_t oldest, bool skip_maps)
   ceph_assert(min <= service.map_cache.cached_key_lower_bound());
 }
 
+int OSD::trim_stale_maps()
+{
+  int num_removed = 0;
+  ObjectStore::Transaction t;
+  for (epoch_t current_epoch = 0;
+       current_epoch < superblock.oldest_map;
+       current_epoch++) {
```

Comment:
This is going to take a really long time after a cluster has existed for a while.  Old clusters can have epoch numbers in the 100k or higher.  Instead, use collection_list to enumerate all of the maps actually stored and compare them against the superblock.